pydo/manager.py

```diff
@@ -330,6 +361,113 @@ def add(
             task_attributes['title'],
         )
 
+    def modify(
+        self,
+        id,
+        title=None,
+        agile=None,
+        body=None,
+        due=None,
+        estimate=None,
+        fun=None,
+        state=None,
+        priority=None,
+        project_id=None,
+        tags=[],
+        tags_rm=[],
+        value=None,
+        willpower=None,
+    ):
+        """
+        Use parent method to modify an existing task
+
+        Arguments:
+            agile (str): Task agile state.
+            title (str): Title of the task.
+            body (str): Description of the task.
+            estimate (float): Estimate size of the task.
+            fun (int): Fun size of the task.
+            state (str): State of the task once it's closed
+            priority (int): Task priority.
+            project_id (str): Project id.
+            tags (list): List of tag ids.
+            tags_rm (list): List of tag ids to remove.
+            title (str): Title of the task.
+            value (int): Objective/Bussiness value of the task.
+            willpower (int): Willpower consumption of the task.
+        """
+        if len(id) < 10:
+            open_tasks = self.session.query(Task).filter_by(state='open')
+            open_task_fulids = [task.id for task in open_tasks]
+            self.log.debug(open_task_fulids)
+            self.log.debug(id)
+            id = self.fulid.sulid_to_fulid(id, open_task_fulids)
+
+        task = self.session.query(Task).get(id)
+        task_attributes = {}
+
+        # Define Project
+        if project_id is not None:
+            project = self.session.query(Project).get(project_id)
+            if project is None:
+                project = Project(id=project_id, description='')
+                self.session.add(project)
+                self.session.commit()
+            task_attributes['project'] = project
+
+        # Define tags
+        task_attributes['tags'] = task.tags
+
+        for tag_id in tags_rm:
+            tag = self.session.query(Tag).get(tag_id)
+            if tag is None:
+                tag = Tag(id=tag_id, description='')
+                self.session.add(tag)
+                self.session.commit()
+            task_attributes['tags'].remove(tag)
+
+        for tag_id in tags:
+            tag = self.session.query(Tag).get(tag_id)
+            if tag is None:
+                tag = Tag(id=tag_id, description='')
+                self.session.add(tag)
+                self.session.commit()
+            task_attributes['tags'].append(tag)
+
+        # Test the task attributes are into the available choices
+        if agile is not None and \
+                agile not in self.config.get('task.agile.states').split(', '):
+            raise ValueError(
+                'Agile state {} is not between the specified '
+                'by task.agile.states'.format(agile)
+            )
+
+        if agile is not None:
```

Comment:
For the rest of the attributes, it would be better to use a generic setter, so if we add new attributes in the future or udas, we won't need to touch this part of the code.

Reply:
You already have a generic setter in `_update`.

Reply:
Solved in 9b995a4e790b6bbdf5896be50ba736a4d45fa248.